### ISSUE TITLE
Style/RegexpLiteral with AllowInnerSlashes: false does not report escaped slashes

### ISSUE THREAD

**Issue by RST-J** [rubocop-hq/rubocop#6488]
When [Style/RegexpLiteral](https://github.com/rubocop-hq/rubocop/blob/master/lib/rubocop/cop/style/regexp_literal.rb) is configured to treat inner slashes as offense and recommend using `%r{}` instead it does not report escaped inner slashes as offense.

## Expected behavior
Escaped backslashes are reported as offense.

## Actual behavior
Escaped backslashes are not reported as offense (`%r{foo\/bar}` is accepted but only %r{foo/bar} should).

## Steps to reproduce the problem
Run rubocop on a regexp in `%r{}` syntax with escaped backslashes while `Style/RegexpLiteral` is configured with `AllowInnerSlashes: false`.

## RuboCop version
```
$ rubocop -V
0.60.0 (using Parser 2.5.3.0, running on ruby 2.5.1 x86_64-linux)
```
